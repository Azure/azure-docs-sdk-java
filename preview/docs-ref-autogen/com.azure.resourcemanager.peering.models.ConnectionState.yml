### YamlMime:JavaType
uid: "com.azure.resourcemanager.peering.models.ConnectionState"
fullName: "com.azure.resourcemanager.peering.models.ConnectionState"
name: "ConnectionState"
nameWithType: "ConnectionState"
summary: "Defines values for ConnectionState."
inheritances:
- "<xref href=\"java.lang.Object\" data-throw-if-not-resolved=\"False\" />"
- "<xref href=\"com.azure.core.util.ExpandableStringEnum\" data-throw-if-not-resolved=\"False\" />"
inheritedMembers:
- "com.azure.core.util.ExpandableStringEnum.<T>fromString(java.lang.String,java.lang.Class<T>)"
- "com.azure.core.util.ExpandableStringEnum.<T>values(java.lang.Class<T>)"
- "com.azure.core.util.ExpandableStringEnum.equals(java.lang.Object)"
- "com.azure.core.util.ExpandableStringEnum.hashCode()"
- "com.azure.core.util.ExpandableStringEnum.toString()"
- "java.lang.Object.clone()"
- "java.lang.Object.finalize()"
- "java.lang.Object.getClass()"
- "java.lang.Object.notify()"
- "java.lang.Object.notifyAll()"
- "java.lang.Object.wait()"
- "java.lang.Object.wait(long)"
- "java.lang.Object.wait(long,int)"
syntax: "public final class ConnectionState extends ExpandableStringEnum<ConnectionState>"
constructors:
- "com.azure.resourcemanager.peering.models.ConnectionState.ConnectionState()"
fields:
- "com.azure.resourcemanager.peering.models.ConnectionState.ACTIVE"
- "com.azure.resourcemanager.peering.models.ConnectionState.APPROVED"
- "com.azure.resourcemanager.peering.models.ConnectionState.NONE"
- "com.azure.resourcemanager.peering.models.ConnectionState.PENDING_APPROVAL"
- "com.azure.resourcemanager.peering.models.ConnectionState.PROVISIONING_COMPLETED"
- "com.azure.resourcemanager.peering.models.ConnectionState.PROVISIONING_FAILED"
- "com.azure.resourcemanager.peering.models.ConnectionState.PROVISIONING_STARTED"
- "com.azure.resourcemanager.peering.models.ConnectionState.VALIDATING"
methods:
- "com.azure.resourcemanager.peering.models.ConnectionState.fromString(java.lang.String)"
- "com.azure.resourcemanager.peering.models.ConnectionState.values()"
type: "class"
metadata: {}
package: "com.azure.resourcemanager.peering.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-peering:1.0.0-beta.1
