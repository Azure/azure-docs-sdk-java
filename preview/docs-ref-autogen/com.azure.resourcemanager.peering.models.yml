### YamlMime:JavaPackage
uid: "com.azure.resourcemanager.peering.models"
fullName: "com.azure.resourcemanager.peering.models"
name: "com.azure.resourcemanager.peering.models"
summary: "Package containing the data models for PeeringManagementClient. Peering Client."
classes:
- "com.azure.resourcemanager.peering.models.BgpSession"
- "com.azure.resourcemanager.peering.models.CdnPeeringPrefixListResult"
- "com.azure.resourcemanager.peering.models.CheckServiceProviderAvailabilityInput"
- "com.azure.resourcemanager.peering.models.ConnectionState"
- "com.azure.resourcemanager.peering.models.ContactDetail"
- "com.azure.resourcemanager.peering.models.DirectConnection"
- "com.azure.resourcemanager.peering.models.DirectPeeringFacility"
- "com.azure.resourcemanager.peering.models.DirectPeeringType"
- "com.azure.resourcemanager.peering.models.Enum0"
- "com.azure.resourcemanager.peering.models.ExchangeConnection"
- "com.azure.resourcemanager.peering.models.ExchangePeeringFacility"
- "com.azure.resourcemanager.peering.models.Family"
- "com.azure.resourcemanager.peering.models.Kind"
- "com.azure.resourcemanager.peering.models.LearnedType"
- "com.azure.resourcemanager.peering.models.LegacyPeeringsKind"
- "com.azure.resourcemanager.peering.models.OperationDisplayInfo"
- "com.azure.resourcemanager.peering.models.OperationListResult"
- "com.azure.resourcemanager.peering.models.PeerAsnListResult"
- "com.azure.resourcemanager.peering.models.PeeringBandwidthOffer"
- "com.azure.resourcemanager.peering.models.PeeringListResult"
- "com.azure.resourcemanager.peering.models.PeeringLocationListResult"
- "com.azure.resourcemanager.peering.models.PeeringLocationPropertiesDirect"
- "com.azure.resourcemanager.peering.models.PeeringLocationPropertiesExchange"
- "com.azure.resourcemanager.peering.models.PeeringLocationsDirectPeeringType"
- "com.azure.resourcemanager.peering.models.PeeringLocationsKind"
- "com.azure.resourcemanager.peering.models.PeeringPropertiesDirect"
- "com.azure.resourcemanager.peering.models.PeeringPropertiesExchange"
- "com.azure.resourcemanager.peering.models.PeeringReceivedRouteListResult"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsnListResult"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefixListResult"
- "com.azure.resourcemanager.peering.models.PeeringServiceCountryListResult"
- "com.azure.resourcemanager.peering.models.PeeringServiceListResult"
- "com.azure.resourcemanager.peering.models.PeeringServiceLocationListResult"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefixEvent"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefixListResult"
- "com.azure.resourcemanager.peering.models.PeeringServiceProviderListResult"
- "com.azure.resourcemanager.peering.models.PeeringServiceSku"
- "com.azure.resourcemanager.peering.models.PeeringSku"
- "com.azure.resourcemanager.peering.models.PrefixValidationState"
- "com.azure.resourcemanager.peering.models.ProvisioningState"
- "com.azure.resourcemanager.peering.models.ResourceTags"
- "com.azure.resourcemanager.peering.models.Role"
- "com.azure.resourcemanager.peering.models.SessionAddressProvider"
- "com.azure.resourcemanager.peering.models.SessionStateV4"
- "com.azure.resourcemanager.peering.models.SessionStateV6"
- "com.azure.resourcemanager.peering.models.Size"
- "com.azure.resourcemanager.peering.models.Tier"
- "com.azure.resourcemanager.peering.models.ValidationState"
interfaces:
- "com.azure.resourcemanager.peering.models.CdnPeeringPrefix"
- "com.azure.resourcemanager.peering.models.CdnPeeringPrefixes"
- "com.azure.resourcemanager.peering.models.LegacyPeerings"
- "com.azure.resourcemanager.peering.models.Operation"
- "com.azure.resourcemanager.peering.models.Operations"
- "com.azure.resourcemanager.peering.models.PeerAsn"
- "com.azure.resourcemanager.peering.models.PeerAsn.Definition"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages.Blank"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages.WithPeerAsn"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages.WithPeerContactDetail"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages.WithPeerName"
- "com.azure.resourcemanager.peering.models.PeerAsn.DefinitionStages.WithValidationState"
- "com.azure.resourcemanager.peering.models.PeerAsn.Update"
- "com.azure.resourcemanager.peering.models.PeerAsn.UpdateStages"
- "com.azure.resourcemanager.peering.models.PeerAsn.UpdateStages.WithPeerAsn"
- "com.azure.resourcemanager.peering.models.PeerAsn.UpdateStages.WithPeerContactDetail"
- "com.azure.resourcemanager.peering.models.PeerAsn.UpdateStages.WithPeerName"
- "com.azure.resourcemanager.peering.models.PeerAsn.UpdateStages.WithValidationState"
- "com.azure.resourcemanager.peering.models.PeerAsns"
- "com.azure.resourcemanager.peering.models.Peering"
- "com.azure.resourcemanager.peering.models.Peering.Definition"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.Blank"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithDirect"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithExchange"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithKind"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithLocation"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithPeeringLocation"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithResourceGroup"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithSku"
- "com.azure.resourcemanager.peering.models.Peering.DefinitionStages.WithTags"
- "com.azure.resourcemanager.peering.models.Peering.Update"
- "com.azure.resourcemanager.peering.models.Peering.UpdateStages"
- "com.azure.resourcemanager.peering.models.Peering.UpdateStages.WithTags"
- "com.azure.resourcemanager.peering.models.PeeringLocation"
- "com.azure.resourcemanager.peering.models.PeeringLocations"
- "com.azure.resourcemanager.peering.models.PeeringReceivedRoute"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.Definition"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.DefinitionStages"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.DefinitionStages.Blank"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.DefinitionStages.WithAsn"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.DefinitionStages.WithParentResource"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.Update"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.UpdateStages"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredAsn.UpdateStages.WithAsn"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.Definition"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.DefinitionStages"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.DefinitionStages.Blank"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.DefinitionStages.WithParentResource"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.DefinitionStages.WithPrefix"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.Update"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.UpdateStages"
- "com.azure.resourcemanager.peering.models.PeeringRegisteredPrefix.UpdateStages.WithPrefix"
- "com.azure.resourcemanager.peering.models.PeeringService"
- "com.azure.resourcemanager.peering.models.PeeringService.Definition"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.Blank"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithLocation"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithPeeringServiceLocation"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithPeeringServiceProvider"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithProviderBackupPeeringLocation"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithProviderPrimaryPeeringLocation"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithResourceGroup"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithSku"
- "com.azure.resourcemanager.peering.models.PeeringService.DefinitionStages.WithTags"
- "com.azure.resourcemanager.peering.models.PeeringService.Update"
- "com.azure.resourcemanager.peering.models.PeeringService.UpdateStages"
- "com.azure.resourcemanager.peering.models.PeeringService.UpdateStages.WithTags"
- "com.azure.resourcemanager.peering.models.PeeringServiceCountries"
- "com.azure.resourcemanager.peering.models.PeeringServiceCountry"
- "com.azure.resourcemanager.peering.models.PeeringServiceLocation"
- "com.azure.resourcemanager.peering.models.PeeringServiceLocations"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.Definition"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.DefinitionStages"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.DefinitionStages.Blank"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.DefinitionStages.WithCreate"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.DefinitionStages.WithParentResource"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.DefinitionStages.WithPeeringServicePrefixKey"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.DefinitionStages.WithPrefix"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.Update"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.UpdateStages"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.UpdateStages.WithPeeringServicePrefixKey"
- "com.azure.resourcemanager.peering.models.PeeringServicePrefix.UpdateStages.WithPrefix"
- "com.azure.resourcemanager.peering.models.PeeringServiceProvider"
- "com.azure.resourcemanager.peering.models.PeeringServiceProviders"
- "com.azure.resourcemanager.peering.models.PeeringServices"
- "com.azure.resourcemanager.peering.models.Peerings"
- "com.azure.resourcemanager.peering.models.Prefixes"
- "com.azure.resourcemanager.peering.models.ReceivedRoutes"
- "com.azure.resourcemanager.peering.models.RegisteredAsns"
- "com.azure.resourcemanager.peering.models.RegisteredPrefixes"
- "com.azure.resourcemanager.peering.models.ResourceProviders"
metadata: {}
package: "com.azure.resourcemanager.peering.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-peering:1.0.0-beta.1
