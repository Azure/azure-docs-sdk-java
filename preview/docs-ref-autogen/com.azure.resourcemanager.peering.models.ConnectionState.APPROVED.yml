### YamlMime:JavaMember
uid: "com.azure.resourcemanager.peering.models.ConnectionState.APPROVED*"
fullName: "com.azure.resourcemanager.peering.models.ConnectionState.APPROVED"
name: "APPROVED"
nameWithType: "ConnectionState.APPROVED"
members:
- uid: "com.azure.resourcemanager.peering.models.ConnectionState.APPROVED"
  fullName: "com.azure.resourcemanager.peering.models.ConnectionState.APPROVED"
  name: "APPROVED"
  nameWithType: "ConnectionState.APPROVED"
  summary: "Static value Approved for ConnectionState."
  syntax: "public static final ConnectionState APPROVED"
type: "field"
metadata: {}
package: "com.azure.resourcemanager.peering.models"
artifact: com.azure.resourcemanager:azure-resourcemanager-peering:1.0.0-beta.1
